akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -306,6 +306,67 @@ object GraphStages {
     override def toString: String = s"SingleSource($elem)"
   }
 
+  final class FutureFlattenSource[T, M](
+    val future: Future[Graph[SourceShape[T], M]])
+    extends GraphStageWithMaterializedValue[SourceShape[T], Future[M]] {
+
+    ReactiveStreamsCompliance.requireNonNullElement(future)
+
+    val out = Outlet[T]("futureFlatten.out")
+    val shape = SourceShape(out)
+
+    override def initialAttributes = DefaultAttributes.futureSource // TODO
+
+    def createLogicAndMaterializedValue(attr: Attributes): (GraphStageLogic, Future[M]) = {
+      val materialized = Promise[M]()
+
+      val logic = new GraphStageLogic(shape) with InHandler with OutHandler {
+        private val sinkIn = new SubSinkInlet[T]("FlattenMergeSink")
+
+        private val cb = getAsyncCallback[Try[Graph[SourceShape[T], M]]] {
+          case scala.util.Success(graph) ⇒ {
+            setHandler(out, this)
+            sinkIn.setHandler(this)
+
+            sinkIn.pull()
```

Comment:
At this point, the `future: Future[Graph[SourceShape[T], M]]` has just completed and this callback is fired from the first out handler: 

```
setHandler(out, new OutHandler {
  def onPull(): Unit =
    future.onComplete(cb)(ExecutionContexts.sameThreadExecutionContext)
})
```

So for me there is a pull, delayed by the prior and required completion of the sub graph.